test/integration/db_integration_test.go

```diff
@@ -64,33 +68,38 @@ func (a *dbIntegrationSuite) TestIntegrationWithDb_GetRecords() {
 	reply, err := stream.CloseAndRecv()
 	require.NoError(a.T(), err)
 	require.True(a.T(), reply.Ok)
-	require.Len(a.T(), a.c.Importer.GetSavedRecords(), recordsCount) // because recordsWithDifferencePulses generates 3 records
+	require.Len(a.T(), records, pulsesNumber)
+
+	jetDrops := make([]types.PlatformJetDrops, 0)
+	for _, r := range records {
+		jetDrop := types.PlatformJetDrops{Records: []*exporter.Record{r}}
+		jetDrops = append(jetDrops, jetDrop)
+	}
+	require.Len(a.T(), jetDrops, pulsesNumber)
 
-	ctx := context.Background()
-	jetDrops := a.be.Extractor().GetJetDrops(ctx)
 	refs := make([]types.Reference, 0)
-	counter := 0
-	for counter < recordsCount {
-		select {
-		case jd := <-jetDrops:
-			transform, err := transformer.Transform(ctx, jd)
-			if err != nil {
-				a.T().Logf("error transforming record: %v", err)
-				return
-			}
-			for _, t := range transform {
-				refs = append(refs, t.MainSection.Records[0].Ref)
-			}
-			counter++
-		case <-time.After(1000 * time.Millisecond):
-			a.T().Fatalf("Timeout waiting for records: expected %v, got %v, saved in importer %v",
-				recordsCount, counter, len(a.c.Importer.GetSavedRecords()))
+	ctx := context.Background()
+	for _, jd := range jetDrops {
+		transform, err := transformer.Transform(ctx, &jd)
+		if err != nil {
+			a.T().Logf("error transforming record: %v", err)
+			return
+		}
+		for _, t := range transform {
+			r := t.MainSection.Records
+			require.NotEmpty(a.T(), r)
+			ref := r[0].Ref
+			require.NotEmpty(a.T(), ref)
+			refs = append(refs, ref)
 		}
 	}
+	require.Len(a.T(), refs, pulsesNumber)
 
-	a.waitRecordsCount(recordsCount)
+	// last record with the biggest pulse number won't be processed, so we do not expect this record in DB
+	expRecordsCount := pulsesNumber - recordsInPulse
```

Comment:
actually you cannot calculate `expRecordsCount` as subtraction to `recordsInPulse `, just use `pulsesNumber -1` because the last pulse didn't change,  that's why

Reply:
Yes we can calculate, this is the exact number of unprocessed records in the last pulse. 
`recordsInPulse=1` in the test. If the declared value will be changed there is no need to change this line.

Reply:
OK change it to 
`expRecordsCount := recordsInPulse*(pulsesNumber - 1)`